PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/ConnectionStringCheck.java

```diff
@@ -0,0 +1,120 @@
+package com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool;
+
+import com.google.gson.JsonParseException;
+import com.intellij.codeInspection.LocalInspectionTool;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiElementVisitor;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiReferenceExpression;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class extends the LocalInspectionTool to check for the use of connection strings in the code.
+ * If the method is called from an Azure client class, a problem is registered with the suggestion message.
+ */
+public class ConnectionStringCheck extends LocalInspectionTool {
+
+
+    /**
+     * This method builds the visitor for the inspection tool.
+     * @param holder ProblemsHolder to register problems
+     * @param isOnTheFly boolean to check if the inspection is on the fly. If true, the inspection is performed as you type.
+     * @return PsiElementVisitor visitor to inspect elements in the code
+     */
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new ConnectionStringVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * This class extends the JavaElementVisitor to visit the elements in the code.
+     * It checks if the method call is a connection string call and if the class is an Azure client.
+     * If both conditions are met, a problem is registered with the suggestion message.
+     */
+    public static class ConnectionStringVisitor extends JavaElementVisitor {
```

Comment:
I was curious if this needs to be public since it is returned as a PsiElementVisitor rather than ConnectionStringVisitor class itself.  Can it be package-private?

Reply:
yes -- I've been changing these to package-private